kafka_enforcer_common/src/main/java/com/tesla/data/enforcer/EnforceCommand.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2019. Tesla Motors, Inc. All rights reserved.
+ */
+
+package com.tesla.data.enforcer;
+
+import static tesla.shade.com.google.common.base.Preconditions.checkNotNull;
+import static tesla.shade.com.google.common.base.Preconditions.checkState;
+
+import com.beust.jcommander.Parameter;
+import com.beust.jcommander.Parameters;
+import io.prometheus.client.Counter;
+import io.prometheus.client.Gauge;
+import io.prometheus.client.exporter.HTTPServer;
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.KafkaAdminClient;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.function.Supplier;
+
+@Parameters(commandDescription = "Enforce given configuration")
+public class EnforceCommand<T> extends BaseCommand<T> {
+
+  private static final Gauge lastSuccess = Gauge.build()
+      .name("kafka_enforcer_run_last_success_timestamp")
+      .help("Timestamp of last successful run as epoch seconds.")
+      .register();
+  private static final Counter failures = Counter.build()
+      .name("kafka_enforce_run_failures_total")
+      .help("Total failures so far.")
+      .register();
+  private static final Counter runs = Counter.build()
+      .name("kafka_enforcer_run_total")
+      .help("Total runs so far.")
+      .register();
+
+  @Parameter(
+      names = {"--dryrun", "-d"},
+      description = "do a dry run")
+  protected boolean dryrun = false;
+  @Parameter(
+      names = {"--unsafemode"},
+      description = "run in unsafe mode, deletion is _only_ allowed in this mode")
+  protected boolean unsafemode = false;
+  @Parameter(
+      names = {"--continuous", "-c"},
+      description = "run enforcement continuously")
+  protected boolean continuous = false;
+  @Parameter(
+      names = {"--interval", "-i"},
+      description = "run interval for continuous mode in seconds",
+      arity = 1)
+  protected int interval = 600;
+
+  private Enforcer<T> enforcer;
+
+  protected EnforceCommand() {
+    // DO NOT REMOVE, this is needed by jcommander
+  }
+
+  // for testing
```

Comment:
nit: not clear how to populate enforcer for non-testing sitatuations. or is it just visible for testing and really would be protected?

Reply:
Do you mean who calls `initEnforcer`? Jcommander will create an instance of this through no arg constructor and then call the `run` method, which would invoke `initEnforcer` -- which is supposed to be overridden. I guess more docs are needed.

Reply:
Added some comments, in a follow up PR, I am going to try making it an abstract class. I recall JCommander didn't let me do so, but will dig deeper so that the abstraction layer is more clear.